src/components/notifications/NotificationsLayout/index.tsx

```diff
@@ -1,14 +1,18 @@
-import React, { Fragment } from 'react'
+import React, { Fragment, useContext } from 'react'
 import { Navigate, Outlet, useLocation } from 'react-router-dom'
 import AppSelector from '../AppSelector'
 import { AnimatePresence } from 'framer-motion'
 import { motion } from 'framer-motion'
+import W3iContext from '../../../contexts/W3iContext/context'
 
 const NotificationsLayout: React.FC = () => {
+  const { activeSubscriptions } = useContext(W3iContext)
   const { pathname } = useLocation()
 
   if (pathname === '/notifications') {
-    return <Navigate to="/notifications/new-app" />
+    if (!activeSubscriptions.length) {
```

Comment:
Why do we do this? One thing about the subscriptions we should be consider: Is it's returning an empty array in the first render and this might lead us. 

So eventually in the first time it'll be empty even though I have subscriptions, should we handle this differently?

Reply:
That's okay for now - maybe down the line we could do something to do with the spinners. 

Maybe something like

- Go to app selector
- Have spinners show up for the subs
- If no subs exist, redirect to discover page
- if they do, show them

Reply:
@devceline Also we are doing the same redirection here: https://github.com/WalletConnect/web3inbox/blob/main/src/components/notifications/AppSelector/index.tsx#L59-L63

Seems like you added mobile condition. Is this redirection expected only on mobile? Should we remove the `useEffect` here?

Reply:
Excellent catch, fixed up the logic. Removed the redirect in `AppSelector`